operator_csv_libs/catalog.py

```diff
@@ -281,7 +323,23 @@ def add_channel_entry(self, channel, name, skiprange=None, replaces=None):
                 if 'entries' in c:
                     c['entries'] = []
                 c['entries'].append(data)
-                
+    
+    #This function adds a channel and bundles from a formatted dict that is provided as a parameter
+    #It expects the same format as is returned by the get_channel_and_bundles function
+    def add_channel_and_bundles(self, channel_and_bundles):
+        #Sanity check on the channel and bundles input parameters
+        if 'channel' not in channel_and_bundles.keys():
+            raise CatalogError("No channel was provided")
+        if 'bundles' not in channel_and_bundles.keys():
+            raise CatalogError("No bundles were provided")
+        
```

Comment:
Need a check to see if the added channel already exists. I'm assuming this is intended to add a new channel with potentially new bundles. Might be worth it to check the bundles and only append the ones that don't already exist as well.